geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala

```diff
@@ -0,0 +1,174 @@
+    val llGh = GeoHash(minX, minY, chunkPrec)
+    val urGh = GeoHash(maxX, maxY, chunkPrec)
+    val (llX, llY, urX, urY) = (llGh.getPoint.getX, llGh.getPoint.getY, urGh.getPoint.getX, urGh.getPoint.getY)
+    val (deltaX, deltaY) = ((llGh.bbox.ur.getX - llGh.bbox.ll.getX), (llGh.bbox.ur.getY - llGh.bbox.ll.getY))
+    val (stepsX, stepsY) = (Math.ceil((urX - llX) / deltaX).toInt, Math.ceil((urY - llY) / deltaY).toInt)
+
+    new File(outDir).mkdir
+
+    val indexes =
+      (for {
+        i <- 0 to stepsX
+        j <- 0 to stepsY
+      } yield (i, j)).par
+    indexes.tasksupport = new ForkJoinTaskSupport(new scala.concurrent.forkjoin.ForkJoinPool(parLevel))
+
+    indexes.foreach { case (i, j) =>
+      val geoHash = GeoHash(llX + deltaX * i, llY + deltaY * j, chunkPrec)
+      getChunkBounds(geoHash, bbox, gridStepX, gridStepY).foreach { case (minX, maxX, minY, maxY) =>
+        if (maxX > minX && maxY > minY) {
+          val cmd = Seq("gdal_translate", "-q", "-of", "GTIFF", "-projwin", s"${minX}", s"${maxY}",
+            s"${maxX}", s"${minY}", file, s"${outDir}/gh_${geoHash.hash}.tif")
+          cmd.!!
+        }
+      }
+    }
+  }
+
+  //Get bounds from given GeoHash that are used to crop raster
+  def getChunkBounds(gh: GeoHash,
+                     bbox: BoundingBox,
+                     stepX: Double,
+                     stepY: Double): Option[(Double, Double, Double, Double)] = {
+    BoundingBox.intersects(bbox, gh.bbox) match {
+      case false => None
+      case _ =>
+        val (ghMinX, ghMinY, ghMaxX, ghMaxY) = (gh.bbox.ll.getX, gh.bbox.ll.getY, gh.bbox.ur.getX, gh.bbox.ur.getY)
+        val (rMinX, rMinY, rMaxX, rMaxY) = (bbox.ll.getX, bbox.ll.getY, bbox.ur.getX, bbox.ur.getY)
+        Some(getCropLocation(ghMinX, rMinX, rMaxX, stepX, false),
+             getCropLocation(ghMaxX, rMinX, rMaxX, stepX, true),
+             getCropLocation(ghMinY, rMinY, rMaxY, stepY, false),
+             getCropLocation(ghMaxY, rMinY, rMaxY, stepY, true))
+    }
+  }
+
+  //Get location on raster grid for cropping operation
+  def getCropLocation(coor: Double,
+                      minCoor: Double,
+                      maxCoor: Double,
+                      step: Double,
+                      isMax: Boolean): Double = {
+    val steps = ((coor - minCoor) / step).toInt
+    val calCoor = {
+      val tmpCoor = minCoor + step * steps.toDouble
+      //NO DUPLICATION: With the following line, adjacent chunks have no overlapped edge.
+      if (coor - tmpCoor > 0.5D * step) tmpCoor + step
+      //DUPLICATION POSSIBLE: With the following line, adjacent chunks may have one edge
+      //(width = 1) overlapped.
+//      if (isMax && coor - tmpCoor > 0D) tmpCoor + step
+      else tmpCoor
+    }
+    Math.min(Math.max(calCoor, minCoor), maxCoor)
+  }
+}
+
+object RasterChunking extends Logging {
+  val BOUND_MIN_X = -180.0
+  val BOUND_MIN_Y = -90.0
+  val BOUND_MAX_X = 180.0
+  val BOUND_MAX_Y = 89.999999
```

Comment:
Yes....   But in order for that to work, GeoHash(180.0, 90.0) needs to do something sensible.  (I told Jianping to do this.)